datafusion/optimizer/src/unwrap_cast_in_comparison.rs

```diff
@@ -428,6 +452,31 @@ fn try_cast_literal_to_type(
     }
 }
 
+/// Cast a timestamp value from one unit to another
+fn cast_between_timestamp(from: DataType, to: DataType, value: i128) -> Option<i64> {
+    let seconds = match from {
```

Comment:
I think it can be improved. if the cast Second to Second, then the code will do unneccessary mul by 1m and then div  by 1m